cypress/support/iterator.js

```diff
@@ -0,0 +1,41 @@
+import config from './config/services';
+import alwaysTests from '../integration/pages';
+
+const serviceHasPageType = (service, pageType) =>
+  config[service].pageTypes[pageType].path !== undefined;
+
+const smokeTest = (service, pageType) =>
+  Cypress.env('SMOKE') ? config[service].pageTypes[pageType].smoke : true;
+
+const iterator = (pageType, tests, canonicalOnlyTests, ampOnlyTests) => {
+  Object.keys(config)
+    .filter(service => serviceHasPageType(service, pageType))
+    .filter(service => smokeTest(service, pageType))
+    .forEach(service => {
+      describe(`${pageType} - ${service} - Canonical`, () => {
+        before(() => {
+          cy.visit(config[service].pageTypes[pageType].path, {
+            failOnStatusCode: !pageType.includes('error'),
+          });
+        });
+
+        if (alwaysTests) alwaysTests({ service, pageType });
+        if (tests) tests({ service, pageType });
```

Comment:
I feel like the naming of these could make this easier to follow - do we mean `always` in the sense of tests that are run for both AMP and canonical pages? Yet they contain page type exclusions, for example https://github.com/bbc/simorgh/pull/3053/files#diff-6a133ac9ddaa5ea2078f57fc5c2567e1R25-R35. And `tests` doesn't really describe what these are.

Also how/why could `alwaysTests` be falsy?